src/test/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRuleTest.java

```diff
@@ -0,0 +1,106 @@
+package org.jenkinsci.test.acceptance.recorder;
+
+import org.hamcrest.core.Is;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import java.io.File;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+public class TestRecorderRuleTest {
+
+    @Test
+    public void shouldNotRecordSuccessTestExecutionByDefault() {
+        TestRecorderRule testRecorderRule = new TestRecorderRule();
+        Description shouldNotRecordSuccessTestExecutionByDefault = Description.createTestDescription("org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest", "shouldNotRecordSuccessTestExecutionByDefault");
+        testRecorderRule.starting(shouldNotRecordSuccessTestExecutionByDefault);
+
+        System.out.println("Hello World");
+
+        testRecorderRule.succeeded(shouldNotRecordSuccessTestExecutionByDefault);
+        testRecorderRule.finished(shouldNotRecordSuccessTestExecutionByDefault);
+
+        File outputFile = new File("target", "org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest-shouldNotRecordSuccessTestExecutionByDefault.mov");
+        assertThat(outputFile.exists(), is(false));
+
+        //Clean the field
+        outputFile.delete();
+    }
+    @Test
+    public void shouldRecordFailingTestExecutionByDefault() {
+
+        TestRecorderRule testRecorderRule = new TestRecorderRule();
+        Description shouldNotRecordSuccessTestExecutionByDefault = Description.createTestDescription("org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest", "shouldRecordFailingTestExecutionByDefault");
+        testRecorderRule.starting(shouldNotRecordSuccessTestExecutionByDefault);
+
+        System.out.println("Good Bye World");
+        //succeeded is not called since a failure is simulated
+        //testRecorderRule.succeeded(shouldNotRecordSuccessTestExecutionByDefault);
+        testRecorderRule.finished(shouldNotRecordSuccessTestExecutionByDefault);
+
+        File outputFile = new File("target", "org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest-shouldRecordFailingTestExecutionByDefault.mov");
+        assertThat(outputFile.exists(), is(true));
+
+        //Clean the field
+        outputFile.delete();
+    }
+
+    @Test
+    public void shouldRecordSuccessTestExecutionWhenSaveAll() {
+        String oldValue = System.getProperty("RECORDER_SAVE_ALL");
+        System.setProperty("RECORDER_SAVE_ALL", "true");
```

Comment:
Yes this is the normal behaviour . It records the screen since this is the only thing we can do now. Maybe with more effort on the project we would be able to detect the browser window, but I think it could be complicated to do it.

Moreover this project basically takes an screenshot every period of time and finally compose a video, not sure if you can record a minimized window.

I am not sure if turning off by default recording feature could be a good approach for adoption or not.